gateway/middleware.go

```diff
@@ -703,34 +703,34 @@ func (t BaseMiddleware) CheckSessionAndIdentityForValidKey(originalKey string, r
 
 	// Only search in RPC if it's not in emergency mode
 	t.Logger().Debug("Querying authstore")
+
 	// 2. If not there, get it from the AuthorizationHandler
 	session, found = t.Spec.AuthManager.SessionDetail(t.Spec.OrgID, key, false)
-	if found {
-		key = session.KeyID
-
-		session := session.Clone()
```

Comment:
Feels like a potential issue to me, which can cause some concurency related issues. 
We use clone for the purpose, so multiple threads in gw can operate with same session objects (clones), so we will not have issues like map writes from multiple threads and etc. 

I think the valid code here is: `session = session.Clone()`